pkg/kube/inject/webhook.go

```diff
@@ -1163,13 +1174,25 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 		// application container's value. At the same time, if user explicitly configures a RunAsUser in the istio-proxy
 		// container which is different to the application container's value, that setting is still honored.
 		if sideCarProxy := FindSidecar(params.pod); sideCarProxy != nil && sideCarProxy.SecurityContext != nil {
-			if isSidecarUserMatchingAppUser(params.pod.Spec.Containers) {
+			if isSidecarUserMatchingAppUser(params.pod) {
 				log.Infof("Resetting the UserID of sideCar proxy as it matches with the app container for Pod %q", params.pod.Name)
 				sideCarProxy.SecurityContext.RunAsUser = nil
 				sideCarProxy.SecurityContext.RunAsGroup = nil
 			}
 		}
 	}
+
+	var nodes kclient.Client[*corev1.Node]
+	if wh.nodes != nil {
+		nodes = wh.nodes.ForCluster(cluster.ID(clusterID))
+		if nodes == nil {
+			log.Warnf("unable to fetch nodes, failed to get client for %q", clusterID)
+		}
+	} else if features.EnableNativeSidecars {
+		log.Warnf("Native sidecars feature is enabled but node client wasn't initialized")
```

Comment:
nit: native sidecars being on or off isn't the real issue here. If the nodes informer isn't initialized, that means the code is incorrect so we want to warn regardless